storage/aptosdb/src/state_store/mod.rs

```diff
@@ -176,42 +176,42 @@ impl StateStore {
         Ok(result)
     }
 
+    /// Read the value for a specific version and require its existence. Call `get_value_by_value`
+    /// for most use cases.
     pub fn get_value_at_version(
         &self,
         key_and_version: &(StateKey, Version),
     ) -> Result<StateValue> {
         self.db
-            .get::<StateValueSchema>(key_and_version)?
-            .ok_or_else(|| {
-                format_err!(
-                    "Canot find value of key {:?} at version {}",
-                    key_and_version.0,
-                    key_and_version.1
-                )
+            .get::<StateValueSchema>(key_and_version)
+            .and_then(|v| {
+                v.ok_or_else(|| format_err!("State Value is missing for {:?}", *key_and_version))
             })
     }
 
-    /// Get the state value given the state key and root hash of state Merkle tree by using the
```

Comment:
Part of the comment is still relevant, can you update it instead of deleting it?